### ISSUE TITLE
False negative clone-on-copy

### ISSUE THREAD

**Issue by piotr-maks** [rust-lang/rust-clippy#5436]
Hi folks,
I ran this code under Clippy:
```rust
fn foo() {
    let mut vec = Vec::new();
    let key = (0, 0);

    vec.push(key.clone());

    let _ = key.clone();
}
```
Clippy produced this error:
```
error: using `clone` on a `Copy` type
 --> src/lib.rs:7:13
  |
7 |     let _ = key.clone();
  |             ^^^^^^^^^^^ help: try removing the `clone` call: `key`
  |
  = note: `-D clippy::clone-on-copy` implied by `-D warnings`
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#clone_on_copy
```
I expected to see same error in `vec.push(key.clone())` but Clippy is silent about it. After removing `.clone()` in line 5 it does compile fine, so I assume it is redundant. 

Can someone explain to me what is the difference here?

Output of `cargo clippy -V`
`clippy 0.0.212 (4ee1206 2020-02-01)`
